src/lib.rs

```diff
@@ -985,6 +985,9 @@ pub trait Itertools : Iterator {
     /// It works similarly to *.skip(* `n` *)* except it is eager and
     /// preserves the iterator type.
     ///
+    /// This will produce incorrect results if the iterator is not *fused* and
+    /// the number of elements in the iterator is less than `n`.
+    ///
```

Comment:
Ok, sorry, can we add another correction? Dropping will work fine if n is larger than the number of elements in the iterator. The error would be to try to get more items from the returned iterator if n is larger than the number of elements.